### ISSUE TITLE
Crash: Product list filter

### ISSUE THREAD

**Issue by 0nko** [woocommerce/woocommerce-android#3032]
**To Reproduce**
1. Go to Products
2. Click on Filters
3. The app crashes

**Stack trace**
```
Process: com.woocommerce.android.prealpha, PID: 7217
    android.content.res.Resources$NotFoundException: String resource ID #0x0
        at android.content.res.Resources.getText(Resources.java:444)
        at android.content.res.Resources.getString(Resources.java:537)
        at android.content.Context.getString(Context.java:645)
        at com.woocommerce.android.viewmodel.ResourceProvider.getString(ResourceProvider.kt:12)
        at com.woocommerce.android.ui.products.ProductFilterListViewModel.buildFilterListItemUiModel(ProductFilterListViewModel.kt:179)
        at com.woocommerce.android.ui.products.ProductFilterListViewModel.loadFilters(ProductFilterListViewModel.kt:77)
        at com.woocommerce.android.ui.products.ProductFilterListFragment.setupObservers(ProductFilterListFragment.kt:140)
        at com.woocommerce.android.ui.products.ProductFilterListFragment.onViewCreated(ProductFilterListFragment.kt:60)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:332)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1199)
        at androidx.fragment.app.FragmentManager.addAddedFragments(FragmentManager.java:2236)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2009)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1965)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1861)
        at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Comment by anitaa1990** [comment]
@nbradbury, this crash looks to be because we added a new product type `OTHER` in `ProductType` enum without a string resource. The app is expecting a string resource in [ProductFilterListViewModel](https://github.com/woocommerce/woocommerce-android/blob/3f8fdd83e5bf7f72ef66a921531cb52533b29202/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductFilterListViewModel.kt#L177-L183) but it's not available. 

Since we are iterating the values of `ProductType` enum, we would also be displaying the `OTHER` product type in product filter list, which I don't think we need to do. Would you be able to take a look at this one?

**Comment by nbradbury** [comment]
@anitaa1990 Yep, I'll tackle this one.

**Comment by nbradbury** [comment]
Ugh, sorry about this crash. I should've tested the product filters after adding `ProductType.OTHER`.

**Comment by nbradbury** [comment]
Oops, looks like @0nko beat me to it: #3033

**Comment by anitaa1990** [comment]
Closed via #3033
